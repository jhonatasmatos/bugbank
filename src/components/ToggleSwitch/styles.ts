import styled, { css } from 'styled-components';

type Props = {
  isChecked: boolean;
};

export const Container = styled.label`
  ${({ theme }) => css`
    border: 1px solid ${theme.colors.primary};
    width: 5rem;
    height: 2rem;
    border-radius: 1rem;
    display: flex;
    align-items: center;
    justify-content: center;
    cursor: pointer;
    position: relative;
    background: ${theme.colors.gray};

    ${(p: Props) =>
      p.isChecked &&
      css`
        background: ${theme.colors.primary};
      `}
  `}
`;

export const Input = styled.input`
  width: 100%;
  height: 100%;
  opacity: 0;
  cursor: pointer;
`;

export const Span = styled.span`
  ${({ theme }) => css`
    position: absolute;
    width: 2rem;
    height: 2rem;
    border-radius: 1rem;
    background: ${theme.colors.white};
    left: 0;
    cursor: pointer;
    box-shadow: 2px 2px 4px ${theme.colors.primary};
    transition: all 0.3s;

    ${(p: Props) =>
      p.isChecked &&
      css`
        left: 60%;
      `}
  `}
`;
