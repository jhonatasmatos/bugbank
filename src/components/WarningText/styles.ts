import styled, { css } from 'styled-components';

interface IWarnigWrapper {
  color?: 'white' | 'secondary';
}

export const WarnigWrapper = styled.p`
  ${({ theme }) => css`
    padding: 1.6rem;
    border: 1px solid ${theme.colors.secondary};
    border-radius: 0.8rem;
    color: ${theme.colors.secondary};
    font-style: italic;
    font-size: 1.3rem;
    text-align: center;
    line-height: 1.6rem;

    span {
      font-weight: bold;
    }

    ${(p: IWarnigWrapper) =>
      p.color === 'white' &&
      css`
        border: 1px solid ${theme.colors.white};
        color: ${theme.colors.white};
      `}
  `}
`;
