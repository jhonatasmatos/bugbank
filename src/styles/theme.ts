export const theme = {
  colors: {
    primary: '#D62A92',
    secondary: '#A422E3',
    tertiary: '#E224ED',
    gray: '#d3d3d3',
    black: '#171717',
    white: '#fff',
  },
  font: {
    family: {
      default: "'Nunito', sans-serif",
    },
  },
  sizes: {
    xsmall: '8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
    xhuge: '6.4rem',
  },
  media: {
    lteMedium: '(max-width: 768px)',
  },
  spacings: {
    xsmall: '8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
    xhuge: '6.4rem',
  },
};
