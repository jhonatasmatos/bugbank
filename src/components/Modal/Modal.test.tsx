import { screen, fireEvent } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { Modal } from '.';

const fn = jest.fn();

describe('<Modal />', () => {
  it('should render success modal', () => {
    const { container, getByText } = renderTheme(
      <Modal
        type="ok"
        text="A conta 123-4 foi criada com sucesso"
        onClose={fn}
      />,
    );

    const modalText = container.querySelector('p');
    const iconError = screen.queryByTestId('ok-icon') as HTMLImageElement;
    const button = getByText('Fechar');

    fireEvent.click(button);

    expect(iconError).toHaveStyle('color: green;');
    expect(button).toBeInTheDocument();
    expect(fn).toHaveBeenCalledTimes(1);
    expect(modalText).toHaveTextContent('A conta 123-4 foi criada com sucesso');
  });

  it('should render error modal', () => {
    const { container, getByText } = renderTheme(
      <Modal type="error" text="Conta inválida ou inexistente" onClose={fn} />,
    );

    const modalText = container.querySelector('p');
    const iconError = screen.queryByTestId('error-icon') as HTMLImageElement;
    const button = getByText('Fechar');

    fireEvent.click(button);

    expect(iconError).toHaveStyle('color: red;');
    expect(button).toBeInTheDocument();
    expect(fn).toHaveBeenCalledTimes(1);
    expect(modalText).toHaveTextContent('Conta inválida ou inexistente');
  });

  it('should render alert modal', () => {
    const { container, getByText } = renderTheme(
      <Modal
        type="alert"
        text="Funcionalidade em desenvolvimento"
        onClose={fn}
      />,
    );

    const modalText = container.querySelector('p');
    const iconError = screen.queryByTestId('alert-icon') as HTMLImageElement;
    const button = getByText('Fechar');

    fireEvent.click(button);

    expect(iconError).toHaveStyle('color: yellow;');
    expect(button).toBeInTheDocument();
    expect(fn).toHaveBeenCalledTimes(1);
    expect(modalText).toHaveTextContent('Funcionalidade em desenvolvimento');
  });

  it('should render with checked false', () => {
    const { container } = renderTheme(
      <Modal
        type="ok"
        text="A conta 123-4 foi criada com sucesso"
        onClose={fn}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
