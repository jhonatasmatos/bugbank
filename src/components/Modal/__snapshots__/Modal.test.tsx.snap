// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should render with checked false 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100vw;
  height: 100vh;
  position: absolute;
  top: 0;
  left: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: rgb(0,0,0,0.5);
  z-index: 3;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 40rem;
  height: 30rem;
  padding: 2.5rem;
  border-radius: 1.5rem;
  position: relative;
  -webkit-animation-duration: 0.5s;
  animation-duration: 0.5s;
  -webkit-animation-name: slidein;
  animation-name: slidein;
  background: #fff;
}

.c2 {
  height: 2rem;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c2 a {
  font-size: 3rem;
  color: #A422E3;
}

.c3 {
  height: 100%;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c4 {
  font-size: 1.8rem;
  text-align: center;
  color: #A422E3;
  white-space: pre-line;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 18rem;
  height: 5rem;
  border-radius: 0.8rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  color: #fff;
  font-size: 1.6rem;
  background: #A422E3;
  -webkit-filter: drop-shadow(2px 2px 4px rgba(0,0,0,0.4));
  filter: drop-shadow(2px 2px 4px rgba(0,0,0,0.4));
}

.c5:hover {
  opacity: 0.8;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <a
          href="#"
        >
          x
        </a>
      </div>
      <div
        class="c3"
      >
        <svg
          color="green"
          data-testid="ok-icon"
          fill="currentColor"
          height="84"
          stroke="currentColor"
          stroke-width="0"
          style="color: green;"
          viewBox="0 0 24 24"
          width="84"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zm0 18c-4.411 0-8-3.589-8-8s3.589-8 8-8 8 3.589 8 8-3.589 8-8 8z"
          />
          <path
            d="M9.999 13.587 7.7 11.292l-1.412 1.416 3.713 3.705 6.706-6.706-1.414-1.414z"
          />
        </svg>
        <p
          class="c4"
          id="modalText"
        >
          A conta 123-4 foi criada com sucesso
        </p>
        <a
          class="c5"
          id="btnCloseModal"
        >
          Fechar
        </a>
      </div>
    </div>
  </div>
</div>
`;
